### PR TITLE
Added logic for validation only visible fields on the dialog

### PR DESCRIPTION
**Pull request by ProkopovVitaliy**
**Description** (*)

Allows validate only visible fields on the dialog. 

Actual result. 
If we have dialog with required hidden fields and try to submit form, we will face issue:

https://user-images.githubusercontent.com/29776509/115829861-2745fa80-a418-11eb-9a07-694881c0ab78.mp4

It's make some inconvenience because if field are hidden we don't want to validate it.

This PR, add feature - ignore validation for hidden fields. Only visible fields will validate. 

Expected result:

https://user-images.githubusercontent.com/29776509/115830385-e13d6680-a418-11eb-98a6-a548faecd44a.mp4

**Fixed Issues (if relevant)**

1. Fixes magento/magento2-phpstorm-plugin#<issue_number>

**Questions or comments**

**Contribution checklist** (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with integration/functional tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by eduard13** [comment]
@ProkopovVitaliy please add some hints to the PR's description how can this be properly tested.
Thanks.

**Comment by ProkopovVitaliy** [comment]
@eduard13 Description updated. 

Thanks!

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/AbstractDialog.java [hunk 1]
```diff
@@ -119,7 +119,7 @@ protected void clearValidationHighlighting() {
      * @return List[FieldValidationData]
      */
     protected List<FieldValidationData> getFieldsToValidate() {
-        return new LinkedList<>(fieldsValidationsList);
+        return getAvailableForFiltering(fieldsValidationsList);
```
Comment (eduard13) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/AbstractDialog.java:122]: Is there any reason to create a new method? Please move the `getAvailableForFiltering` implementation into this method, so we won't have redundant methods that do the same.

**Review by eduard13 — CHANGES_REQUESTED** [review]
Hi @ProkopovVitaliy, could you please review my suggestion below?
Thank you.

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/AbstractDialog.java (hunk 1 above)
Comment (ProkopovVitaliy) [reply src/com/magento/idea/magento2plugin/actions/generation/dialog/AbstractDialog.java]: @eduard13 Removed redundant method.

**Comment by ProkopovVitaliy** [comment]
@eduard13 Removed redundant method. Ready for review. 

Thanks!

**Review by eduard13 — APPROVED** [review]
Thanks 👍

**Pull request merged**